codalab/worker/bundle_manager.py

```diff
@@ -529,3 +534,25 @@ def _construct_run_message(self, worker, bundle):
         resources['request_network'] = bundle.metadata.request_network
 
         return message
+
+    def _fail_unresponsive_bundles(self):
+        """
+        Fail bundles in uploading, staged and running state if we haven't heard from them for more than
+        BUNDLE_TIMEOUT_DAYS days.
+        """
+        bundles_to_fail = (
+            self._model.batch_get_bundles(state=State.UPLOADING)
+            + self._model.batch_get_bundles(state=State.STAGED)
+            + self._model.batch_get_bundles(state=State.RUNNING)
+        )
+        now = time.time()
+
+        for bundle in bundles_to_fail:
+            if now - bundle.metadata.created > BUNDLE_TIMEOUT_DAYS * SECONDS_PER_DAY:
```

Comment:
Add a comment saying this should really be the last updated time logically, but we are using created for simplicity.